### ISSUE TITLE
Error parsing Java 7' try with resources

### ISSUE THREAD

**Issue by jsotuyod** [cobertura/cobertura#220]
Using Cobertura 2.0.3, on the following snippet:

```
                try (
                        final InputStream in = ProcessEnvironment.class.getClassLoader()
                                .getResourceAsStream("libenvironment.so");
                        final FileOutputStream out = new FileOutputStream(f);
                ) {
```

I get:

> [cobertura] WARN  [main] net.sourceforge.cobertura.reporting.ComplexityCalculator - JavaNCSS got an error while parsing the java file /usr/share/tomcat7/.jenkins/jobs/Tyque - Pipe Process Lib i386/workspace/pipe-process/src/main/java/com/monits/process/ProcessEnvironment.java
ParseException in STDIN
Last useful checkpoint: "com.monits.process.ProcessEnvironment"
Encountered " "final" "final "" at line 93, column 25.
Was expecting one of:
    "assert" ...
    "boolean" ...
    "byte" ...
    "char" ...
    "double" ...
    "enum" ...
    "false" ...
    "float" ...
    "int" ...
    "long" ...
    "new" ...
    "null" ...
    "short" ...
    "super" ...
    "this" ...
    "true" ...
    "void" ...
    <INTEGER_LITERAL> ...
    <FLOATING_POINT_LITERAL> ...
    <CHARACTER_LITERAL> ...
    <STRING_LITERAL> ...
    <IDENTIFIER> ...
    "(" ...
    ")" ...
    "!" ...
    "~" ...
    "++" ...
    "--" ...
    "+" ...
    "-" ...
    
The issue has already been solved almost a year ago in [JavaNCSS 33.54](https://jira.codehaus.org/browse/JAVANCSS-49), but according to Cobertura's changelog, we are using 33.53 since Cobertura 1.9.3

**Comment by dennisl** [comment]
Unfortunately Cobertura does not use JavaNCSS as a dependency, but has made a copy of the classes, which have then been modified. So upgrading to a newer version of JavaNCSS is not a simple task.

Personally I don't see why Cobertura uses JavaNCSS at all. People who wants JavaNCSS reports can use JavaNCSS directly. I think it should be removed from Cobertura.

**Comment by raniejade** [comment]
Right now it prevents us to generate reports (hmtl/xml). I agree that it should be removed, though we should consider why was it included in the first place.

**Comment by jsotuyod** [comment]
Agreed, I'd get rid of it, I just want my coverage reports up and running. If it was really needed, add it as dependency, forking when the upstream project is still under active development is not generally a good idea.

**Comment by shevek** [comment]
Can this be made to work, please? We need cobertura reports as a part of our business process, but our world is increasingly Java7, and Cobertura is dropping out of our projects one by one as we upgrade each one.

**Comment by jsotuyod** [comment]
I just moved away from cobertura and adopted jacoco. The lack on activity on this issue,  and others related to java 7 & 8 proved this project to no longer be viable for my company projects.

**Comment by raniejade** [comment]
Based on 5e365bd2d1083c5223dddbd2f524d124f1871e08, probably the custom is removed? If that's the case probably we can just switch out the bundled classes with the latest ones?

**Comment by christ66** [comment]
@jsotuyod It's not the same issue. The issue you are encountering is because javancss does not understand the `final` keyword.

JavaNCSS is used to calculate the codes complexity when generating reports. There is a roadmap for removing the code complexity calculation if it is not required: https://github.com/cobertura/cobertura/wiki/Roadmap#ability-to-enabledisable-code-complexity-calculator.

**Comment by RockyMM** [comment]
Can you give an estimate when the feature of removing the code complexity calculations will be in Cobertura?
